crates/oxc_linter/src/autofix/fixer.rs

```diff
@@ -1,26 +1,178 @@
-use std::borrow::Cow;
+use std::{borrow::Cow, cmp::Ordering};
+
+use oxc_ast::Span;
 
 use super::Fix;
 
 pub struct Fixer<'a> {
     source_text: &'a str,
-    fixes: Vec<Fix>,
+    fixes: Vec<Fix<'a>>,
 }
 
 impl<'a> Fixer<'a> {
-    pub fn new(source_text: &'a str, fixes: Vec<Fix>) -> Self {
+    pub fn new(source_text: &'a str, mut fixes: Vec<Fix<'a>>) -> Self {
+        fixes.sort_by(
+            |Fix { span: Span { start: a_start, end: a_end }, .. },
+             Fix { span: Span { start: b_start, end: b_end }, .. }| {
+                if a_start < b_start {
+                    Ordering::Less
+                } else if a_start > b_start {
+                    Ordering::Greater
+                } else {
+                    if a_end < b_end {
+                        Ordering::Less
+                    } else if a_end > b_end {
+                        Ordering::Greater
+                    } else {
+                        Ordering::Equal
+                    }
+                }
+            },
+        );
```

Comment:
Can we simplify this by deriving `PartialOrd`/`Ord` on `Span`?

```suggestion
    pub fn new(source_text: &'a str, mut fixes: Vec<Fix<'a>>) -> Self {
        fixes.sort_by_key(|fix| fix.span)
```

Reply:
Thank you for this advice! I'm going to do this.

Reply:
Used derive in [5de4bf9](https://github.com/Boshen/oxc/pull/101/commits/5de4bf960cfa2cd0773f5259518f017313208192)